relay/pkg/chain/ethereum/ethereum.go

```diff
@@ -236,3 +236,27 @@ func (ec *ethereumChain) MarkNewHeaviest(
 
 	return nil
 }
+
+// MarkNewHeaviestPreflight performs a preflight call of the
+// MarkNewHeaviest method to check whether its execution will
+// succeed.
+func (ec *ethereumChain) MarkNewHeaviestPreflight(
+	ancestorDigest [32]byte,
+	currentBestHeader []byte,
+	newBestHeader []byte,
+	limit *big.Int,
+) bool {
+	result, err := ec.relayContract.CallMarkNewHeaviest(
+		ancestorDigest,
+		currentBestHeader,
+		newBestHeader,
+		limit,
+		nil,
+	)
+	if err != nil {
+		logger.Warnf("MarkNewHeaviest preflight failed with: [%v]", err)
+		return false
+	}
```

Comment:
There might be a situation when `CallMarkNewHeaviest` fails for a reason other than the preceding `AddHeader` transaction being not yet mined and in such case, we'll log the problem on warn level instead of logging it on error level. 

Is there any other on-chain check we can perform instead of doing a preflight call? Can we use one of the view functions of [the contract](https://github.com/summa-tx/relays/blob/master/solidity/contracts/Relay.sol)?

Reply:
If the problem persists, the whole process will eventually fail (after reaching max attempts) on error level and cause a restart of the forwarder. The same logic was used in the legacy relay maintainer. Regarding view functions, that needs to be investigated, hard to say at the moment. However, even if it's possible, it will be more complicated as it will probably require to call multiple functions (to cover all the preconditions checks of `MarkNewHeaviest`). We will have to deal with possible errors from calls as well and decide what to do with them, in the same way as we're currently doing here.

Reply:
Can we backlog a task to do this investigation? If the contract does not allow for a one-call check, doing a preflight is better from the perspective of Ethereum client calls bandwidth costs. But if there is a way to do one call, I think it is better than the preflight.

Reply:
Sure, added to the backlog.